src/Compilers/CSharp/Portable/Symbols/Source/SourceExtensionTypeSymbol.cs

```diff
@@ -61,6 +61,17 @@ protected override void CheckUnderlyingType(BindingDiagnosticBag diagnostics)
                 var location = singleDeclaration.NameLocation;
 
                 underlyingType.CheckAllConstraints(DeclaringCompilation, conversions, location, diagnostics);
+
+                if (!IsExplicitExtension)
+                {
+                    foreach (var typeParameter in TypeParameters)
```

Comment:
>TypeParameters [](http://example.com/codeflow?start=50&length=14)

Should we check containing type type parameters as well?

Reply:
No. The type parameters from the containing type should be considered "locked" (not subject to substitution). I'll add that missing check in `CanImplicitlyExtend`.